ddtrace/contrib/httplib/__init__.py

```diff
@@ -58,9 +58,10 @@
 
 :ref:`Headers tracing <http-headers-tracing>` is supported for this integration.
 """
-from .patch import get_version
-from .patch import patch
-from .patch import unpatch
+
+from ..internal.httplib.patch import get_version
+from ..internal.httplib.patch import patch
+from ..internal.httplib.patch import unpatch
```

Comment:
Why do we not need to do
```
from . import patch as _ # noqa: F401, I001
```
for this one?

Reply:
it allows users to import from `ddtrace.contrib.httplib.patch` directly

Reply:
Good catch. We should add it here. 

We honestly got lucky here. If we swap the order of these two imports [from ddtrace.contrib.httplib.patch import should_skip_request](https://github.com/DataDog/dd-trace-py/blob/da67b1f50f58539aab8f681d166413500797d8ad/tests/contrib/httplib/test_httplib.py#L14C30-L14C35) AND [from ddtrace.contrib.httplib import patch](https://github.com/DataDog/dd-trace-py/blob/da67b1f50f58539aab8f681d166413500797d8ad/tests/contrib/httplib/test_httplib.py#L12C1-L12C42) then most `httplib` tests will fail with the following error: `patch() E               TypeError: 'module' object is not callable`. The order we import the patch module and the patch function matter. We need to fix this module. 

We should also keep this mind for all of our integrations. We may not have explicit test coverage for this case in every integration 

Also it would we be nice if this module used required_modules helper in this integration. Something like this:

```
from ...internal.utils.importlib import require_modules


required_modules = ["http.client"]

with require_modules(required_modules) as missing_modules:
    if not missing_modules:
        from . import patch as _ # noqa: F401, I001
        ......
```

Reply:
Yeah I was wondering about that too. Is it just an older integration, so it pre-dates `require_modules()` and didn't get it?

Reply:
Ya that was a miss on our part and I am afraid if we don't update it now we may never get to it

Reply:
ok got it! will update and push soon